File: composeApp/src/commonMain/java/com/github/catvod/net/OkhttpInterceptor.java
package com.github.catvod.net;

import com.github.catvod.utils.Utils;
import okhttp3.*;
import okio.BufferedSource;
import okio.Okio;

import java.io.IOException;
import java.util.zip.Inflater;
import java.util.zip.InflaterInputStream;

public class OkhttpInterceptor implements Interceptor {

    @Override
    public Response intercept(Chain chain) throws IOException {
        Response response = chain.proceed(getRequest(chain));
        String encoding = response.header("Content-Encoding");
        if (response.body() == null || encoding == null || !encoding.equals("deflate")) return response;
        InflaterInputStream is = new InflaterInputStream(response.body().byteStream(), new Inflater(true));
        return response.newBuilder().headers(response.headers()).body(new ResponseBody() {
            @Override
            public MediaType contentType() {
                return response.body().contentType();
            }

            @Override
            public long contentLength() {
                return response.body().contentLength();
            }

            @Override
            public BufferedSource source() {
                return Okio.buffer(Okio.source(is));
            }
        }).build();
    }

    private Request getRequest(Chain chain) {
        Request request = chain.request();
        if (request.url().host().equals("gitcode.net"))
            return request.newBuilder().addHeader("User-Agent", Utils.CHROME).build();
        return request;
    }
}

File: composeApp/src/commonMain/kotlin/com/github/catvod/net/OkhttpInterceptor.kt
package com.github.catvod.net

import io.ktor.client.plugins.*
import okhttp3.ResponseBody
import okio.BufferedSource
import okio.buffer
import okio.source
import java.io.IOException


class OkhttpInterceptor : okhttp3.Interceptor {
    @Throws(IOException::class)
    override fun intercept(chain: okhttp3.Interceptor.Chain): okhttp3.Response {
        val response = chain.proceed(getRequest(chain))
        val encoding = response.header("Content-Encoding")
        if (encoding == null || (encoding != "deflate")) return response
        val `is` = java.util.zip.InflaterInputStream(response.body.byteStream(), java.util.zip.Inflater(true))
        return response.newBuilder().headers(response.headers).body(object : ResponseBody() {
            override fun contentType(): okhttp3.MediaType? {
                return response.body.contentType()
            }

            override fun contentLength(): Long {
                return response.body.contentLength()
            }

            override fun source(): BufferedSource {
                return `is`.source().buffer()
            }
        }).build()
    }

    private fun getRequest(chain: okhttp3.Interceptor.Chain): okhttp3.Request {
        val request = chain.request()
//        if (request.url.host == "gitcode.net") return request.newBuilder().addHeader("User-Agent", UserAgent.)
//            .build()
        return request
    }
}
